### PR TITLE
Feat/signin signup pages

### PR DESCRIPTION
**Pull request by vlzelensky**

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import api from 'config/axios';
+import { LoginResponse } from 'api/types';
+
+export const registerUser = async (userData: {
+  name: string;
```
Comment (dariausatova) [inline src/api/index.ts]: use interface

File: src/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  password: string;
+  repeatPassword: string;
+  birthDate: string;
+  email: string;
+}) => {
+  try {
+    await api().post('user/register', userData);
+    return false;
+  } catch (e) {
+    return true;
```
Comment (dariausatova) [inline src/api/index.ts]: true? not a mistake?

File: src/api/index.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+};
+
+export const logIn = (userData: { email: string; password: string }) => {
+  try {
+    return api()
+      .post<LoginResponse>('user/login', userData)
+      .then((res) => res.data);
+  } catch (e) {
+    console.error(e);
```
Comment (dariausatova) [inline src/api/index.ts]: use error handlers

File: src/api/index.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+};
+
+export const getUser = (token: string) => {
+  return api()
+    .get<LoginResponse>('user/me', { headers: { Authorization: token } })
+    .then((res) => res.data)
+    .catch((e) => {
+      console.warn(e);
```
Comment (dariausatova) [inline src/api/index.ts]: use error handlers

File: src/components/Form/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import { FC } from 'react';
+import { Form as AntdForm, Input, Button, DatePicker, message } from 'antd';
```
Comment (dariausatova) [inline src/components/Form/Form.tsx:2]: more than 5 elements use the transfer to different lines

File: src/components/Form/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import { Link, useNavigate } from 'react-router-dom';
+import { registerUser } from 'api';
+import { user } from 'store/user';
+import { FormProps } from './types';
+import { validatePassword, validateRepeatPassword, validateUserName } from './validators';
```
Comment (dariausatova) [inline src/components/Form/Form.tsx]: use the transfer to different lines

File: src/components/Form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+import './styles.css';
+
+export const Form: FC<FormProps> = ({ title, mode }) => {
+  const navigate = useNavigate();
+
+  const isSignUp = mode === 'signup';
+
+  const onFinish = async (values: {
+    name: string;
```
Comment (dariausatova) [inline src/components/Form/Form.tsx]: use interface

File: src/components/Form/Form.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    userName: string;
+    password: string;
+    repeatPassword: string;
+    birthDate: string;
+    email: string;
+  }) => {
+    if (isSignUp) {
+      const isError = await registerUser(values);
+      if (isError) {
+        message.error('Пользователь с таким e-mail уже существует');
```
Comment (dariausatova) [inline src/components/Form/Form.tsx]: return message.error('Пользователь с таким e-mail уже существует');

File: src/components/Form/Form.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        return;
+      }
+      message.success('Вы успешно зарегистрированы');
+      setTimeout(() => navigate('/signin'), 1500);
+      return;
+    }
+    await user.signIn({ email: values.email, password: values.password });
```
Comment (dariausatova) [inline src/components/Form/Form.tsx]: the result does not need to be processed and used?

Comment (dariausatova) [inline src/components/Form/Form.tsx]: use destructurization

File: src/components/Form/validators.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import { FormRule } from 'antd';
+
+export const validatePassword = (isSignUp: boolean) => (_: FormRule, value: string) => {
+  if (isSignUp) {
+    const pattern = new RegExp(
+      /(?=[A-Za-z0-9@#$%^&+!=]+$)^(?=.*[A-Z])(?=.*[0-9])(?=.{8,}).*$/g
+    );
+    if (!value || (value.length > 8 && pattern.test(value))) {
```
Comment (dariausatova) [inline src/components/Form/validators.ts:8]: can't if/else be used?

Comment (dariausatova) [inline src/components/Form/validators.ts:8]: and why !value? can it be empty?

File: src/components/Form/validators.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  return Promise.resolve();
+};
+
+export const validateRepeatPassword = ({
+  getFieldValue,
+}: {
+  getFieldValue: (name: string) => string;
+}) => ({
+  validator(_: FormRule, value: string) {
+    if (!value || getFieldValue('password') === value) {
```
Comment (dariausatova) [inline src/components/Form/validators.ts:24]: higher

File: src/components/Form/validators.ts [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      return Promise.resolve();
+    }
+    return Promise.reject(new Error('Пароли не совпадают'));
+  },
+});
+
+export const validateUserName = (_: FormRule, value: string) => {
+  if (value && value.indexOf(' ') >= 0) {
```
Comment (dariausatova) [inline src/components/Form/validators.ts:32]: use trim

File: src/store/user.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
+import { makeAutoObservable } from 'mobx';
+import { logIn, getUser } from 'api';
+import { initApi } from 'config/axios';
+
+class User {
+  constructor() {
+    makeAutoObservable(this);
+  }
```
Comment (dariausatova) [inline src/store/user.ts:8]: add empty line

File: src/store/user.ts [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  async signIn(userData: { password: string; email: string }) {
+    const data = await logIn(userData);
+    this.isAuth = true;
+    this.email = data!.email;
+    this.name = data!.name;
+    this.userName = data!.userName;
+    this.birthDate = data!.birthDate;
+    localStorage.setItem('token', data!.token);
+    initApi(data!.token);
+  }
```
Comment (dariausatova) [inline src/store/user.ts:29]: add empty line

File: src/components/Form/Form.tsx (hunk 5 above)
Comment (vlzelensky) [reply src/components/Form/Form.tsx:2]: there are exactly 5 elements

File: src/components/Form/validators.ts (hunk 10 above)
Comment (vlzelensky) [reply src/components/Form/validators.ts:8]: if there is no value it returns Promise.resolve() cause there is another error handler that runs in this case

File: src/components/Form/validators.ts (hunk 11 above)
Comment (vlzelensky) [reply src/components/Form/validators.ts:24]: if there is no value it returns Promise.resolve() cause there is another error handler that runs in this case

File: src/components/Form/validators.ts (hunk 12 above)
Comment (vlzelensky) [reply src/components/Form/validators.ts:32]: why? It just warn user not to use whitespaces in userName, it doesn't delete any of them

File: src/components/Form/styles.css [hunk 15]
```diff
@@ -0,0 +1,15 @@
+.form-wrapper {
+    max-width: 400px;
```
Comment (dariausatova) [inline src/components/Form/styles.css]: why 4 characters indent?

**Pull request merged**